### PR TITLE
Implemented sending multiple requests via WC

### LINKED ISSUE #64: batch transactions - via additional method
**Is your feature request related to a problem? Please describe.**
https://github.com/hgraph-io/hedera-walletconnect-utils/issues/10 provides a method for batch transactions.

This ticket provides an alternate solution by allowing the first param in the `request.params` array to be an array of method bodies (such as a transaction body). An example of an update looks like:

``` typescript
public parseSessionRequest(
    event: Web3WalletTypes.SessionRequest,
    // optional arg to throw error if request is invalid, call with shouldThrow = false when calling from rejectSessionRequest as we only need id and top to send reject response
    shouldThrow = true,
  ): {
    ...
    body?: Transaction | Transaction[] | Query<any> | Query<any>[] | Uint8Array[] | undefined
    accountId?: AccountId
```

**Describe the solution you'd like**
Add a method that processes multiple transactions on the network (similar to `hedera_signTransactionAndSend`)

### PR DESCRIPTION
**Pull request by BeeJeeNinja**
Closes #64

### CODE REVIEW

File: src/lib/shared/utils.ts [hunk 1]
```diff
@@ -70,9 +70,9 @@ export function base64StringToTransaction<T extends Transaction>(transactionByte
  * @param Query
  * @returns a base64 encoded string
  */
-export function queryToBase64String<T, Q extends Query<T>>(query: Q): string {
-  const queryBytes = query.toBytes()
-  return Buffer.from(queryBytes).toString('base64')
+export function queryToBase64String<T, Q extends Query<T> = Query<T>>(query: Q): string {
+  const queryBytes = query.toBytes();
+  return Buffer.from(queryBytes).toString('base64');
```
Comment (hgraphql) [inline src/lib/shared/utils.ts:75]: @BeeJeeNinja please elaborate on these changes

File: src/lib/shared/utils.ts [hunk 2]
```diff
@@ -83,9 +83,9 @@ export function queryToBase64String<T, Q extends Query<T>>(query: Q): string {
  * @param queryBytes string - a base64 encoded string
  * @returns `Query<T>`
  */
-export function base64StringToQuery<T, Q extends Query<T>>(queryBytes: string): Q {
-  const decoded = Buffer.from(queryBytes, 'base64')
-  return Query.fromBytes(decoded) as Q
+export function base64StringToQuery<T, Q extends Query<T> = Query<T>>(queryBytes: string): Q {
+  const decoded = Buffer.from(queryBytes, 'base64');
+  return Query.fromBytes(decoded) as Q;
```
Comment (hgraphql) [inline src/lib/shared/utils.ts:88]: @BeeJeeNinja same here, why these changes?

File: src/lib/wallet/index.ts [hunk 3]
```diff
@@ -101,18 +101,15 @@ export default class Wallet extends Web3Wallet implements HederaNativeWallet {
     })
   }
 
-  /*
-   *  Session Requests
-   */
   public parseSessionRequest(
```
Comment (hgraphql) [inline src/lib/wallet/index.ts:104]: I think we could turn this into a recursive function and test the first param of these JSON-RPC methods to test whether or not it's an array

Comment (hgraphql) [reply src/lib/wallet/index.ts:104]: we'd then update the <parse|execute>SessionRequest functions to support this, then we'd be able able to pass along multiple requests of same method type without making to many modifications to parts of the code

**Review by hgraphql — COMMENTED** [review]
added some comments

File: src/lib/shared/utils.ts (hunk 1 above)
Comment (BeeJeeNinja) [reply src/lib/shared/utils.ts:75]: well, I guess we don't really need this

**Comment by hgraphql** [comment]
@BeeJeeNinja closing this PR for now, thank you for providing the example of this way of sending multiple requests. We will use it for further discussion with the project to determine how we want to proceed with multiple transaction functionality.
